source/retryable-reads/tests/README.rst

```diff
@@ -0,0 +1,269 @@
+commands listed in the "failCommands" field. See `SERVER-35004`_ and
+`SERVER-35083`_ for more information.
+
+.. _SERVER-35004: https://jira.mongodb.org/browse/SERVER-35004
+.. _SERVER-35083: https://jira.mongodb.org/browse/SERVER-35083
+
+The ``failCommand`` fail point may be configured like so::
+
+    db.adminCommand({
+        configureFailPoint: "failCommand",
+        mode: <string|document>,
+        data: {
+          failCommands: ["commandName", "commandName2"],
+          closeConnection: <true|false>,
+          errorCode: <Number>,
+        }
+    });
+Some tests depend on a server fail point, ``failCommand``, which
+allows us to force a network error before the server would return a read result
+to the client.  See `SERVER-35004`_ for
+more information.
+
+.. _SERVER-29606: https://jira.mongodb.org/browse/SERVER-35004
+
+The fail point may be configured like so::
+
+    db.runCommand({
+        configureFailPoint: "failCommand",
+        mode: <string|document>,
+        data: <document>
+    });
+    
+``mode`` is a generic fail point option and may be assigned a string or document
+value. The string values ``"alwaysOn"`` and ``"off"`` may be used to enable or
+disable the fail point, respectively. A document may be used to specify either
+``times`` or ``skip``, which are mutually exclusive:
+
+- ``{ times: <integer> }`` may be used to limit the number of times the fail
+  point may trigger before transitioning to ``"off"``.
+- ``{ skip: <integer> }`` may be used to defer the first trigger of a fail
+  point, after which it will transition to ``"alwaysOn"``.
+
+The ``data`` option is a document that may be used to specify options that
+control the fail point's behavior. ``failCommand`` supports the following
+``data`` options, which may be combined if desired:
+
+- ``failCommands``: Required, the list of command names to fail.
+- ``closeConnection``: Boolean option, which defaults to ``false``. If
+  ``true``, the connection on which the command is executed will be closed
+  and the client will see a network error.
+- ``errorCode``: Integer option, which is unset by default. If set, the
+  specified command error code will be returned as a command error.
+
+Disabling Fail Point after Test Execution
+-----------------------------------------
+
+After each test that configures a fail point, drivers should disable the
+``failCommands`` fail point to avoid spurious failures in
+subsequent tests. The fail point may be disabled like so::
+
+    db.runCommand({
+        configureFailPoint: "failCommands",
+        mode: "off"
+    });
+
+Network Error Tests
+===================
+
+Network error tests are expressed in YAML and should be run against a standalone,
+shard cluster, or single-node replica set. Until `SERVER-34943`_ is resolved,
+these tests cannot be run against a multi-node replicaset.
+
+
+.. _SERVER-34943: https://jira.mongodb.org/browse/SERVER-34943
+
+Test Format
+-----------
+
+Test Format
```

Comment:
It looks like the actual test files use a different format with more fields than what is documented here. For example, failCommand_failPoint, retryable_command_started_event and retryable_operation. This format seems different from the retryable writes format. Can you document these fields and how they are used?

Reply:
As discussed offline, I'll refactor the anchors I used so that these fields no longer exist as top-level fields.